spec/contracts/placeholder_users/create_contract_spec.rb

```diff
@@ -0,0 +1,41 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2021 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+require 'contracts/shared/model_contract_shared_context'
+
+describe PlaceholderUsers::CreateContract do
+  include_context 'ModelContract shared context'
+
+  let(:placeholder_user) { FactoryBot.create(:placeholder_user) }
```

Comment:
The model for a `CreateContract` should not be persisted. Ideally, the `PlaceholderUser.new` method is used and attributes (name) assigned to the newly instantiated model.

Every available attribute should then be speced for validations. In this case, this would be that the type be `"PlaceholderUser"` and that the name be present, potentially also the length restriction.

To avoid duplication, part of it might be extracted into a shared examples file to be used by the specs for the CreateContract and UpdateContract.

The specs on contracts need to be among the most thorough of the application as it in part already is and in other will more and more become the central authority for security and data integrity.

Reply:
Done.